### ISSUE TITLE
firebase_auth plugin does not send silent push notifications to iOS devices for phone authentication.  Always uses reCAPTCHA.

### ISSUE THREAD

**Issue by kroikie** [FirebaseExtended/flutterfire#601]
Phone authentication should send a silent push notification to the user whether or not the user has given permission for push notifications.  In reality, the user is prompted with reCAPTCHA every single time, even when on an actual iOS device.

I know my APN keys are setup correctly because when I grant permission for push notifications, all my push notifications from Firebase cloud messaging work fine.  The silent push notifications do not work to prevent reCAPTCHA if the user has not granted push notifications permission.  That is not how it's supposed to work [according to the docs](https://firebase.google.com/docs/auth/ios/phone-auth#start-receiving-silent-notifications).  Users should get silent push notifications regardless of their permissions.

```dart
final FirebaseAuth _auth = FirebaseAuth.instance;

await _auth.verifyPhoneNumber(
        phoneNumber: testPhoneNumber,
        timeout: const Duration(seconds: 5),
        verificationCompleted: verificationCompleted,
        verificationFailed: verificationFailed,
        codeSent: codeSent,
        codeAutoRetrievalTimeout: codeAutoRetrievalTimeout
);
```

After the above code is run on a physical iOS device, the user is sent to a reCAPTCHA verification.

But if later the user grants permission for push notifications, the user is never asked for reCAPTCHA verification.  Silent push notifications are not working as outlined in the docs:

https://firebase.google.com/docs/auth/ios/phone-auth#start-receiving-silent-notifications

**Comment by kroikie** [comment]
@sbefort

The issue at https://github.com/flutter/flutter/issues/23429 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by christianalfoni** [comment]
@kroikie Finally I found the reason why this is not working :-) We have a pretty huge issue with this on our project due to:

"iOS (using Recaptcha) - Phone authentication on iOS will show the project's url instead of the app's name in the verification SMS when using the reCAPTCHA flow. This is expected behavior. The reCAPTCHA flow was added for cases where an APNS token is not available (including on real devices). If APNS is not correctly configured, a real device will default to the reCAPTCHA flow."

The "project url" in Firebase is often blocked by mobile providers. Not sure why, but we have given reports to Google and they report back that the SMS is not sent due to untrusted/invalid URL in the sms.

Only way we are able to solve this is to ask our users to manually send an SMS so that we can create a "test user" with a code we send back to them. 

Thanks for opening up the issue!
